packages/block-library/src/embed/settings.js

```diff
@@ -60,9 +72,29 @@ export function getEmbedBlockSettings( { title, description, icon, category = 'e
 				const { url } = ownProps.attributes;
 				const core = select( 'core' );
 				const { getEmbedPreview, isPreviewEmbedFallback, isRequestingEmbedPreview, getThemeSupports } = core;
-				const preview = undefined !== url && getEmbedPreview( url );
+
+				let preview = false;
+				let fetching = false;
+
+				if ( undefined !== url ) {
+					if ( options.preview ) {
+						// We have a custom preview, so pass it the response from oembed and the attributes
+						// and use whatever it returns.
+						preview = options.preview( getEmbedPreview( url ), ownProps.attributes );
+					} else {
+						preview = getEmbedPreview( url );
+					}
+
+					if ( options.requesting ) {
+						// To support custom previews that use a rendering API endpoint, `options.requesting`
+						// should return if the API request is in progress.
+						fetching = options.requesting( url );
+					} else {
+						fetching = isRequestingEmbedPreview( url );
+					}
+				}
```

Comment:
Another minor note here, but may be worth pulling out to helper functions, so it's easier to reason about the control flow:

```jsx
getPreview = ( ownProps ) => {
    const { url } = ownProps;
    if ( ! url ) {
        return false;
    }
    if ( options.preview ) {
        options.preview( getEmbedPreview( url ), ownProps.attributes );
    }
    return options.preview( getEmbedPreview( url ), ownProps.attributes );   
}
```